packages/iris-grid/src/IrisGridTableModel.ts

```diff
@@ -78,6 +78,24 @@ class IrisGridTableModel
     return this.table.applyCustomColumns != null;
   }
 
+  getMemoizedKeyColumnSet = memoize(
+    (tableColumns: DhType.Column[], inputTableKeys: string[]) =>
+      new Set(
+        tableColumns
+          .filter((column: DhType.Column) =>
+            inputTableKeys.includes(column.name)
+          )
+          .map((column: DhType.Column) => column.name)
+      )
+  );
+
+  get keyColumnSet(): Set<ColumnName> {
+    return this.getMemoizedKeyColumnSet(
+      this.table.columns,
+      this.inputTable?.keys ?? []
```

Comment:
We should use `EMPTY_ARRAY` here instead of `[]`. Looks like we're making that same mistake in a couple other methods below (`frontColumns`).
The problem with falling back to `[]` is that if `inputTable` is undefined, then we're creating a new array each time, which means the memoization is _always_ going to fail, since two different array instances do not equal each other (e.g. `[] == []` is always `false`).

Looks like `backColumns`/`getMemoizedBackColumns` has it (mostly) correct below - pass in `undefined` if it's not defined, and then in the memoized function return an array (though the `getMemoizedBackColumns` method should fallback to returning `EMPTY_ARRAY`, not `[]`).